examples/record.rs

```diff
@@ -0,0 +1,56 @@
+extern crate rbspy;
+
+use rbspy::sampler::{record, RecordConfig};
+use rbspy::OutputFormat;
+
+fn main() {
+    let mut process = std::process::Command::new(path_to_ruby_binary())
+        .arg("ci/ruby-programs/infinite.rb")
+        .spawn()
+        .unwrap();
+    let out_path = std::path::PathBuf::from("rbspy-out.svg");
+
+    let config = RecordConfig {
+        format: OutputFormat::flamegraph,
+        raw_path: std::path::PathBuf::from("rbspy-raw.txt"),
+        out_path: out_path.clone(),
```

Comment:
One thing I'm thinking about is it would be nice to have some sort of a way to exchange these directly in memory and thus avoid overhead when a) serializing profiling data and b) writing it to file.

I don't know how to do that though, especially in a way that would work with a simpler C api.

@AdrK, what do you think? Would it have to be a char* buffer that users provide or something like that?

Reply:
I'll keep this in mind. It seems doable for individual snapshot data, but I don't see an obvious path for recording.

Reply:
Imo, the best would be to use shared memory approach, Linux has a shm lib for this purpose. Shared file is quite slow apporach, as the open, seek and close operations are expensive. If we can assume that writer & reader will always be in the same process, we can simply pass a pointer to the memory between those.

Reply:
Now that the components are separated, we could consider exporting the `sampler` as a way to get low-level access to traces. A program like Pyroscope could include a small shim (probably in Rust) that creates a sampler and starts it up, collecting traces into batches and passing them in memory to C/Go/whatever code that consumes them. I'd prefer not have that complexity in rbspy itself, but it seems reasonable for rbspy's library crate to provide the tracing infrastructure to support it.